src/java/org/apache/cassandra/db/aggregation/AggregationSpecification.java

```diff
@@ -89,45 +106,126 @@ public final GroupMaker newGroupMaker()
     public abstract GroupMaker newGroupMaker(GroupingState state);
 
     /**
-     * Creates a new <code>AggregationSpecification</code> instance that will build aggregates based on primary key
-     * columns.
+     * Creates a new {@code Factory} instance to create {@code AggregationSpecification} that will build aggregates
+     * based on primary key columns.
      *
      * @param comparator the comparator used to compare the clustering prefixes
      * @param clusteringPrefixSize the number of clustering columns used to create the aggregates
-     * @return a new <code>AggregationSpecification</code> instance that will build aggregates based on primary key
-     * columns
+     * @return a  new {@code Factory} instance to create {@code AggregationSpecification} that will build aggregates
+     * based on primary key columns.
      */
-    public static AggregationSpecification aggregatePkPrefix(ClusteringComparator comparator, int clusteringPrefixSize)
+    public static AggregationSpecification.Factory aggregatePkPrefixFactory(ClusteringComparator comparator,
+                                                                            int clusteringPrefixSize)
     {
-        return new AggregateByPkPrefix(comparator, clusteringPrefixSize);
+        return new Factory()
+        {
+            @Override
+            public AggregationSpecification newInstance(QueryOptions options)
+            {
+                return new  AggregateByPkPrefix(comparator, clusteringPrefixSize);
+            }
+        };
```

Comment:
Nit: we can return the more concise form:
```suggestion
        return options -> new  AggregateByPkPrefix(comparator, clusteringPrefixSize);
```